db/hints/manager.cc

```diff
@@ -768,4 +801,62 @@ future<> manager::migrate_ip_directories() {
     co_await io_check(sync_directory, _hints_dir.native());
 }
 
+future<> manager::perform_migration() {
+    // This function isn't marked as noexcept, but the only parts of the code that
+    // can throw an exception are:
+    //   1. the call to `migrate_ip_directories()`: if we fail there, the failure is critical.
+    //      It doesn't lead to any data corruption, but the node must be stopped;
+    //   2. the re-initialization of the endpoint managers: a failure there is the same failure
+    //      that can happen when starting a node. It may be seen as critical, but it should only
+    //      boil down to not initializing some of the endpoint managers. No data corruption
+    //      is possible.
+    if (_state.contains(state::stopping) || _state.contains(state::draining_all)) {
+        co_return;
```

Comment:
It's not. If we're stopping the node, then we drop the idea of migrating hinted handoff – there's neither time nor need for that. The other case is less clear because there is no clear semantics of draining the local node – it's not specified when exactly it happens. However, in practice it only occurs when the node is leaving the cluster, which boils down to the same thing – stopping it.
I can add a comment explaining this in the code.